tests/beta/test_default_qubit_tf.py

```diff
@@ -1011,9 +962,33 @@ def circuit(a):
         expected = tf.sin(a)
         assert np.allclose(grad, expected, atol=tol, rtol=0)
 
-    def test_tf_interface_gradient(self, tol):
-        """Tests that the gradient of the circuit fixture above is correct
-        using the TensorFlow interface"""
+    def test_prob_differentiability(self, tol):
+        """Test that the device probability can be differentiated"""
+        dev = qml.device("default.qubit.tf", wires=2)
+
+        @qml.qnode(dev, diff_method="backprop", interface="tf")
+        def circuit(a, b):
+            qml.RX(a, wires=0)
+            qml.RY(b, wires=1)
+            qml.CNOT(wires=[0, 1])
+            return qml.probs(wires=[1])

```

Comment:
making more and more sense to implement `qml.state` as a new (optional) return object